doc/index.html

```diff
@@ -0,0 +1,170 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
```

Comment:
I'd like to not include the actual generated docs in this PR. Let's get the config.ld and doc comments thoroughly vetted before we commit the docs themselves, to avoid churn over the next few dozen commits/PRs.

Reply:
That makes sense. To enable github pages we'll need to add a new branch specifically for the docs anyway, either that ore make a dedicated repo just for the docs (I like the branch option more). I'll create an example on my clone of the repo.

Reply:
Oh, that's nice. Glad they did that.